File: wear/src/main/java/xyz/klinker/messenger/util/CircularOffsettingHelper.java
package xyz.klinker.messenger.util;

import android.support.wearable.view.DefaultOffsettingHelper;
import android.support.wearable.view.WearableRecyclerView;
import android.view.View;

public class CircularOffsettingHelper extends DefaultOffsettingHelper {
    @Override
    public void updateChild(View child, WearableRecyclerView parent) {
        int progress = child.getTop() / parent.getHeight();
        child.setTranslationX(-child.getHeight() * progress);
    }
}

File: wear/src/main/java/xyz/klinker/messenger/util/CircularOffsettingHelper.kt
package xyz.klinker.messenger.util

import android.support.wearable.view.DefaultOffsettingHelper
import android.support.wearable.view.WearableRecyclerView
import android.view.View

class CircularOffsettingHelper : DefaultOffsettingHelper() {
    override fun updateChild(child: View, parent: WearableRecyclerView) {
        val progress = child.top / parent.height
        child.translationX = (-child.height * progress).toFloat()
    }
}
